core/src/ecs/system/dispatcher.cpp

```diff
@@ -62,11 +68,35 @@ void Dispatcher::tagSetBeforeTag(const std::string& tag)
     mTagSettings[tag]->after.tag.push_back(mCurrTag);
 }
 
+void Dispatcher::groupAddGroup()
+{
+    // CUBOS_WARN("{}", mCurrGroup->getGroupTag());
+    mCurrGroup = mCurrGroup->addGroupStep(mCurrTag, mCurrGroup);
+    // CUBOS_WARN("{}", mCurrGroup->getGroupTag());
+}
+
 void Dispatcher::systemAddTag(const std::string& tag)
 {
     ENSURE_CURR_SYSTEM();
     ENSURE_TAG_SETTINGS(tag);
     mCurrSystem->tags.insert(tag);
+    std::shared_ptr<GroupStep> group = mMainStep->findGroup(tag);
+    if (group != nullptr)
+    {
+        mCurrSystem->grouptag = tag;
```

Comment:
```suggestion
    {
        CUBOS_ASSERT(mCurrSystem->grouptag == nullptr, "Systems can only be tagged directly with at most one repeating tag");
        mCurrSystem->grouptag = tag;
```
We should check this right?